src/components/application_manager/src/commands/mobile/subscribe_vehicle_data_request.cc

```diff
@@ -250,6 +250,15 @@ void SubscribeVehicleDataRequest::on_event(const event_engine::Event& event) {
       result_code = mobile_apis::Result::IGNORED;
       response_info = "Already subscribed on some provided VehicleData.";
     }
+
+    if (!vi_waiting_for_subscribe_.empty()) {
+      LOG4CXX_DEBUG(logger_, "Subscribing to all pending VehicleData");
+      VehicleInfoSubscriptions::const_iterator key =
```

Comment:
Shouldn't there be locks placed around `vi_waiting_for_subscribe_`? its use here and the `vi_waiting_for_subscribe_.insert` below could be called concurrently?